qrand/platforms/qsharp/circuit.py

```diff
@@ -28,76 +28,129 @@
 ###############################################################################
 class QsharpCircuit(QuantumCircuit):
     def __init__(self, num_qubits: int) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+        self.gates: str = ""
+        self._num_qubits: int = num_qubits
+
+    def _validate_qubit_index(self, qubit_index: int) -> None:
+        if qubit_index >= self.num_qubits:
```

Comment:
Could you pass a negative number here?

Reply:
No.

Reply:
You need to validate that too @Avhijit-codeboy. You can import the `validate_natural_number` helper function for it. Remember to set `zero=True` in the arguments.

Reply:
Okay, will do that @pedrorrivero.

Reply:
I am assuming that qubit arrays are zero-indexed in Q#, are they?

Reply:
Yes they are.